File: ServiceCore/Translators/ActivityTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class ActivityTranslator
    {
        public static ErrorCode TranslateEntityToBO(Activity _entity, ActivityBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.ID = _entity.ActivityId;
            bo.Name = _entity.Omschrijving;
            if (_entity.Group != null)
            {
                bo.Group.ID = _entity.Group.GroupId;
                bo.Group.Name = _entity.Group.Name;
                bo.Group.Lot = System.Convert.ToInt16(_entity.Group.Lot);
            }

            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(Activity _entity, ActivityBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.Omschrijving = bo.Name;
            if (bo.Group != null)
                // If _entity.ActivityGroup Is Nothing Then
                // _entity.ActivityGroup = New ActivityGroup()
                // _entity.ActivityGroup.Lot = bo.Group.Lot
                // _entity.ActivityGroup.Name = bo.Group.Name
                // Else
                _entity.GroupId = bo.Group.ID;
            return ErrorCode.Success;
        }
    }
}

File: Service/Translators/ActivityTranslator.vb
Imports BO
Imports DAL

Public Class ActivityTranslator
    Public Shared Function TranslateEntityToBO(_entity As Activity, bo As ActivityBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.ID = _entity.ActivityID
        bo.Name = _entity.Omschrijving
        If _entity.ActivityGroup IsNot Nothing Then
            bo.Group.ID = _entity.ActivityGroup.GroupID
            bo.Group.Name = _entity.ActivityGroup.Name
            bo.Group.Lot = _entity.ActivityGroup.Lot
        End If

        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As Activity, bo As ActivityBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.Omschrijving = bo.Name
        If bo.Group IsNot Nothing Then
            'If _entity.ActivityGroup Is Nothing Then
            '    _entity.ActivityGroup = New ActivityGroup()
            '    _entity.ActivityGroup.Lot = bo.Group.Lot
            '    _entity.ActivityGroup.Name = bo.Group.Name
            'Else
            _entity.GroupID = bo.Group.ID
            'End If

            'If _entity.ActivityGroup.GroupID = 0 Then
            '    _entity.ActivityGroup = New ActivityGroup()
            '    _entity.ActivityGroup.Lot = bo.Group.Lot
            '    _entity.ActivityGroup.Name = bo.Group.Name
            'End If
        End If
        Return ErrorCode.Success
    End Function
End Class
